### DISCUSSION TITLE
Should we enable our Java Platform Module System support by default?

### DISCUSSION THREAD

**Discussion by mnriem** [piranhacloud/piranha#2053] [Roadmap]
Currently for Piranha Server you can enable Java Platform Module System support. 

Should we make that the default and change it from an opt-in to an opt-out?

Thoughts?

@Thihup @arjantijms

**Comment by Thihup** [comment]
So... I think we should start Piranha Server in the module path by default (so instead of the `-jar` it would use the `-p lib -m cloud.piranha.server` in the `run.sh/start.sh`).

One side effect of this is that as we require only a few modules to startup, some modules aren't included by default, like the `java.sql` module. So if a user wants to deploy some app that uses SQL, it would have to somehow add the `--add-modules java.sql` flag to have it included in the set of root modules. 

(What happens is the following: the Java would start resolving all the requires that are listed in the `cloud.piranha.server` module and only what is required actually is resolved, using the `--add-modules java.sql`, it would resolve all the requires of `cloud.piranha.server` and `java.sql`)

However, for the `--jpms` flag, I think we should have it disable by default for now. The name of it is kind of misleading. 
The idea of it is to run each webapp in a separate layer. While it sounds complicated, the result of it is that it adds the module information to the ClassLoader that we specify, and in this case is the Webapp's ClassLoader (Piranha Micro supports this too 😄 )

But as the behavior of how each Webapp should be deployed in a layer is not specified, I don't think would be a good have it enable by default, at least for now.

Nevertheless, from my research, we're the first to offer the second feature (run apps in layers) 😎

**Comment by mnriem** [comment]
@Thihup thanks for the clarification. Can you see if the --jpms can be implemented as a WebApplicationExtension? Note with your explanation regarding the server level startup I am now gravitating more towards it staying as an opt-in as I would not want to surprise our customers with the fact that indeed a `java.sql` is not available by default.

**Comment by nipafx** [comment]
Good conversation and good points by @Thihup. One note: the OpenJDK community tries to get rid of the JPMS moniker, so if any command-line flag is created, naming it something like `--modules`, `--modular`, `--layers`, etc. would be nice. 😃
